PowerVCF.psm1

```diff
@@ -5690,6 +5690,348 @@ Export-ModuleMember -Function Get-VCFWsa
+    .DESCRIPTION
+    The Remove-VCFIdentityProvider cmdlet is used to delete an identity provider by specifying its type as "Embedded" or "Microsoft ADFS".
+    .EXAMPLE
+    Remove-VCFIdentityProvider -type Embedded -domainName sfo.rainpole.io.
+    This example shows how to delete an embedded identity provider with a specific domain name.
+    .EXAMPLE
+    Remove-VCFIdentityProvider -type "Microsoft ADFS".
+    This example shows how to delete an external identity provider.
+    #>
+
+    param(
+    [Parameter(Mandatory = $true)][ValidateSet("Embedded","Microsoft ADFS")] [string]$type,
+    [Parameter(Mandatory = $false)][ValidateNotNullOrEmpty()] [string]$domainName
+    )
+    Try {
+        if ($type -eq "Embedded") {
+            $id = (Get-VCFIdentityProviderIdByType -Type $type)
+            createHeader # Calls createHeader function to set Accept & Authorization
+            checkVCFToken # Calls the CheckVCFToken function to validate the access token and refresh if necessary
+            $uri = "https://$sddcManager/v1/identity-providers/$id/identity-sources/$domainName"
+            Invoke-RestMethod -Method DELETE -Uri $uri -Headers $headers # This API does not return a response
+        }
+        elseif ($type -eq "Microsoft ADFS") {
+            $id = (Get-VCFIdentityProviderIdByType -Type $type)
+            createHeader # Calls createHeader function to set Accept & Authorization
+            checkVCFToken # Calls the CheckVCFToken function to validate the access token and refresh if necessary
+            $uri = "https://$sddcManager/v1/identity-providers/$id"
+            Invoke-RestMethod -Method DELETE -Uri $uri -Headers $headers # This API does not return a response
+        }
+    } Catch {
+        ResponseException -Object $_
+    }
+}
+Export-ModuleMember -Function Remove-VCFIdentityProvider
+
+function New-VCFIdentityProvider {
+    <#
+    .SYNOPSIS
+    Connects to the specified SDDC Manager and configures an identity provider.
+    .DESCRIPTION
+    The New-VCFIdentityProvider cmdlet connects to the specified SDDC Manager and configures an embedded or external identity provider from a JSON spec.
+    .EXAMPLE
+    New-VCFIdentityProvider -type Embedded -json .\embeddedIDPSpec.json
+    This example shows how to configure an embedded identity provider from the JSON spec.
+    .EXAMPLE
+    New-VCFIdentityProvider -type "Microsoft ADFS" -json .\externalIDPSpec.json
+    This example shows how to configure "Microsoft ADFS" identity provider from the JSON spec.
+    #>
+
+    param(
+    [Parameter(Mandatory = $true)][ValidateSet("Embedded","Microsoft ADFS")] [string]$type,
+    [Parameter(Mandatory = $true)][ValidateNotNullOrEmpty()] [string]$json
+    )
+    Try {
+        if ($type -eq "Embedded") {
+            $jsonBody = validateJsonInput -json $json
+            $id = (Get-VCFIdentityProviderIdByType -Type Embedded)
+            createHeader # Calls createHeader function to set Accept & Authorization
```

Comment:
Should be able to place the `createHeader` and `checkVCFToken` just after the `Try` instead of each `if`.